pkg/queue/consumers/interface.go

```diff
@@ -0,0 +1,23 @@
+package consumer
+
+import "time"
+
+type QueueConsumer interface {
+
+	//GetMessages gets tries to get the number of messages mentioned in the param "numOfMessages"
+	//from the previous committed offset. If the available messages in the queue are less, returns
+	// how many ever messages are availble
+	GetMessages(numOfMessages int, timeout time.Duration) ([]string, error)
+
+	//GetUnCommitedMessages returns the messages read in the previous call
+	//to GetMessages - if they are not yet committed.
+	// This is especially usefull in the cases where a client
+	//consuming the messsages crashes and when a new instance comes back
+	//it first needs to call this.
```

Comment:
but when the client comes back again, it may get connected to a different instance, right?

Reply:
Is there a way to get current count of uncommited messages in kafka and use that? instead of locally keeping uncommitted messages?

Reply:
I discussed this with. Lets discuss for more clarity